sdk/iot/core/inc/az_iot_provisioning_client.h

```diff
@@ -0,0 +1,310 @@
+  AZ_IOT_PROVISIONING_CLIENT_STATUS_TIMEOUT = 504,
+} az_iot_provisioning_client_status;
+
+/**
+ * @brief Checks if the status indicates a successful operation.
+ *
+ * @param status The #az_iot_hub_client_status to verify.
+ * @return True if the status indicates success. False otherwise.
+ */
+AZ_NODISCARD bool az_iot_provisioning_client_is_success_status(
+    az_iot_provisioning_client_status status);
+
+/**
+ * @brief Checks if the status indicates a retriable error occurred during the operation.
+ *
+ * @param status The #az_iot_hub_client_status to verify.
+ * @return True if the operation should be retried. False otherwise.
+ */
+AZ_NODISCARD bool az_iot_provisioning_client_is_retriable_status(
+    az_iot_provisioning_client_status status);
+
+/**
+ * @brief Gets the MQTT topic filter to subscribe to register responses.
+ *
+ * @param client
+ * @param mqtt_topic_filter An empty #az_span with sufficient capacity to hold the MQTT topic
+ *                          filter.
+ * @param out_mqtt_topic_filter The output #az_span containing the MQTT topic filter.
+ * @return #az_result
+ */
+AZ_NODISCARD az_result az_iot_provisioning_client_register_subscribe_topic_filter_get(
+    az_iot_provisioning_client const* client,
+    az_span mqtt_topic_filter,
+    az_span* out_mqtt_topic_filter);
+
+/**
+ * @brief The registration operation state.
+ * @note This is returned only when the operation completed.
+ *
+ */
+typedef struct az_iot_provisioning_client_registration_state
+{
+  az_span assigned_hub_hostname; /** @brief Assigned Azure IoT Hub hostname. @note This is only
+                                    available if error_code is success. */
+  az_span device_id; /** @brief Assigned device ID. */
+  az_span json_payload; /** @brief Additional JSON payload. */
+  az_iot_provisioning_client_status error_code; /** @brief The register operation status. */
+  uint32_t extended_error_code; /** @brief The extended, 6 digit error code. */
+  az_span error_message; /** @brief Error description. */
+} az_iot_provisioning_client_registration_state;
+
+/**
+ * @brief Register or query operation response.
+ *
+ */
+typedef struct az_iot_provisioning_client_register_response
+{
+  az_iot_provisioning_client_status status; /** @brief The current request status.
+                                             * @note The response for the register operation is
+                                             * available through #registration_state only. */
+  az_span request_id; /** @brief Request ID for the register operation.
+                       * @note This must match all messages associated with the registration
+                       * operation. */
+  az_span registration_state; /** @brief An #az_span containing the state of the register operation.
+                               * @details This can be one of the following: 'unassigned',
+                               * 'assigning', 'assigned', 'failed', 'disabled'. */
+  uint32_t

```

Comment:
>uint32_t [](start = 2, length = 8)

Nit - this and some other vars here don't align.

Reply:
I'm using the clang auto-formatter. Since the names are long, there aren't a lot of ways to massage this better. Let's take it offline and find a nicer solution.